src/dune_rules/simple_rules.ml

```diff
@@ -216,17 +217,27 @@ let copy_files sctx ~dir ~expander ~src_dir (def : Copy_files.t) =
         Path.build file_dst)
   in
   let+ () =
-    Memo.Option.iter def.alias ~f:(fun alias ->
+    Memo.Option.iter alias ~f:(fun alias ->
         let alias = Alias.make alias ~dir in
         Rules.Produce.Alias.add_deps alias (Action_builder.path_set targets))
   in
   targets
 
+let copy_files sctx ~dir ~expander ~src_dir
+    ({ add_line_directive; alias; mode; files; syntax_version; _ } :
+      Copy_files.t) =
+  copy_files' sctx ~dir ~expander ~src_dir ~add_line_directive ~alias ~mode
+    ~files ~syntax_version
+
 let copy_files sctx ~dir ~expander ~src_dir (def : Copy_files.t) =
   Expander.eval_blang expander def.enabled_if >>= function
   | true -> copy_files sctx ~dir ~expander ~src_dir def
   | false -> Memo.return Path.Set.empty
 
+let extra_files_melange sctx ~dir ~expander ~src_dir files =
```

Comment:
I originally tried to create this function inside `Melange_rules` but got some cyclic dependency error:

```
Error: dependency cycle between modules in _build/default/src/dune_rules:
   Virtual_rules
-> Dir_contents
-> Compilation_context
-> Dir_contents
```